### PR TITLE
support 3.4 API

### PR DESCRIPTION
**Pull request by Eldinnie**
WIP
Live location stuff added See #864

### CODE REVIEW

**Comment by Eldinnie** [comment]
Ok I think I got everything in. Something weird going on with test_official. I don't really get the error. I'm hoping @bomjacob can find time to figure it out.

Also we need to let the bot set a stickerset on our group every test (going to be problematic I'm affraid) to properly test the `set_chat_sticker_set` and `delete_chat_sticker_set` methods.

Review wanted.

**Comment by Eldinnie** [comment]
I found the problem with test_official.
It seems TG has their table messed up on:
https://core.telegram.org/bots/api#inputlocationmessagecontent
and
https://core.telegram.org/bots/api#inlinequeryresultlocation

**Comment by Eldinnie** [comment]
Note from @jh0ker Very nice, although perhaps it would be good to mention in the docstring that you can pass either location or lat/lon to edit_message_live_location

**Comment by Eldinnie** [comment]
Need to up tests aswell. Hopefully have time tomorrow

**Comment by Eldinnie** [comment]
@tsnoam added the requested check (and tests). If we can live with a small decrease in coverage due to not testing the groupstickerset stuff I think it's ready to merge/release

**Comment by Eldinnie** [comment]
@tsnoam 
`if not (latitude is not None or longitude is not None) ^ bool(location):`
Ugly, but works.
If location and either lat or lon is not None it will raise.

**Comment by tsnoam** [comment]
@Eldinnie As discussed, please add () around the entire logical expression which start just after if not.

**Comment by Eldinnie** [comment]
@tsnoam Done

File: telegram/chat.py [hunk 1]
```diff
@@ -38,6 +38,9 @@ class Chat(TelegramObject):
         invite_link (:obj:`str`): Optional. Chat invite link, for supergroups and channel chats.
         pinned_message (:class:`telegram.Message`): Optional. Pinned message, for supergroups.
             Returned only in get_chat.
+        sticker_set_name (:obj:`str`): Optional. For supergroups, name of Group sticker set.
+        can_set_sticker_set	(:obj:`bool`): Optional. ``True``, if the bot can change group the
```
Comment (bomjacob) [inline telegram/chat.py]: I think a tab character snuck in here

File: telegram/chat.py [hunk 2]
```diff
@@ -61,6 +64,10 @@ class Chat(TelegramObject):
         pinned_message (:class:`telegram.Message`, optional): Pinned message, for supergroups.
             Returned only in get_chat.
         bot (:class:`telegram.Bot`, optional): The Bot to use for instance methods.
+        sticker_set_name (:obj:`str`, optional): For supergroups, name of Group sticker set.
+            Returned only in get_chat.
+        can_set_sticker_set	(:obj:`bool`, optional): ``True``, if the bot can change group the
```
Comment (bomjacob) [inline telegram/chat.py]: Same here

File: telegram/chat.py (hunk 1 above)
Comment (Eldinnie) [reply telegram/chat.py]: it did will fix

**Pull request merged**